File: lilac/models/log.go
package models

import "time"

type Log struct {
	ID           uint32 `gorm:"primaryKey"`
	UserID       uint32 `gorm:"not null"`
	Plugin       string `gorm:"index;not null;size:31"`
	Level        int32  `gorm:"not null"`
	Ip           string `gorm:"index;not null;size:45"`
	ResourceType string `gorm:"index;not null;size:127"`
	ResourceID   *uint32
	Message      string    `gorm:"not null"`
	CreatedAt    time.Time `gorm:"not null"`
}

File: dahlia/src/main/kotlin/com/github/saturn_siv/palm/dahlia/plugins/cms/models/Log.kt
package com.github.saturn_siv.palm.dahlia.plugins.cms.models

import jakarta.persistence.*
import java.time.Instant

@Table(name = "logs")
@Entity(name = "log")
class Log(
    @Id
    @GeneratedValue(strategy = GenerationType.IDENTITY)
    @Column(nullable = false)
    var id: Int,
    @Column(nullable = false)
    var plugin: String,
    @Column(nullable = false)
    var ip: String,
    @Enumerated(EnumType.STRING)
    @Column(nullable = false)
    var level: Level,
    @Column(nullable = false)
    var resourceType: String,
    @Column
    var resourceId: Int,
    @Column(nullable = false)
    var message: String,
    @Column(nullable = false)
    var createdAt: Instant,
    @ManyToOne(fetch = FetchType.LAZY, optional = false)
    @JoinColumn(name = "user_id")
    val user: User,
) {
    enum class Level {
        DEBUG, INFO, WARN, ERROR
    }
}